### PR TITLE
Feature/update nuget package icon (closes #22)

### LINKED ISSUE #22: Update Cake.Yaml NuGet package icon to the new Cake-Contrib logo for Cake addins
We've recently released new icons for Cake extensions, to help more easily differentiate Cake Addins, Modules, Recipes, etc.

[cake-contrib logos](https://github.com/cake-contrib/graphics)

More information: https://github.com/cake-contrib/graphics

### PR DESCRIPTION
**Pull request by escarls**
## Description
1. Changed PackageIconUrl to PackageIcon to fix NU5048 as PackageIconUrl is deprecated.
2. Updated the icon to the new graphics provided [here](https://github.com/cake-contrib/graphics)

## Related Issue
Issue #22 

## Motivation and Context
The change seemed like an easy one, so trying to contribute as I'm using the package.

## How Has This Been Tested?
Before the change, the nuget did not contain icon.png and packaging give the NU5048 warning (if building with newer Cake.Recipe than is used today)
After the change, there are no errors or warnings for the icon anymore and the package contains icon.png

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/Cake.Yaml/Cake.Yaml.csproj [hunk 1]
```diff
@@ -14,13 +14,14 @@
     <Authors>Redth</Authors>
     <Owners>Redth,cake-contrib</Owners>
     <PackageProjectUrl>https://github.com/cake-contrib/Cake.Yaml</PackageProjectUrl>
-    <PackageIconUrl>https://cdn.jsdelivr.net/gh/cake-contrib/graphics/png/cake-contrib-medium.png</PackageIconUrl>
+    <PackageIcon>images/icon.png</PackageIcon>
```
Comment (augustoproiete) [inline src/Cake.Yaml/Cake.Yaml.csproj:17]:
```suggestion
    <PackageIcon>images/icon.png</PackageIcon>
    <PackageIconUrl>https://cdn.jsdelivr.net/gh/cake-contrib/graphics/png/addin/cake-contrib-addin-medium.png</PackageIconUrl>
    <NoWarn>$(NoWarn);NU5048</NoWarn>
```

Embedded icon is the way forward for sure, but we also keep the `PackageIconUrl` for devs that use NuGet clients that don't yet support embedded icons. Need to disable the `NU5048` warning as well (_The 'PackageIconUrl'/'iconUrl' element is deprecated_).

**Review by augustoproiete — CHANGES_REQUESTED** [review]

File: src/Cake.Yaml/Cake.Yaml.csproj (hunk 1 above)
Comment (escarls) [reply src/Cake.Yaml/Cake.Yaml.csproj:17]: I've added back the PackageIconUrl.

Comment (escarls) [reply src/Cake.Yaml/Cake.Yaml.csproj:17]: And it seems it's not needed to suppress the NU5048 warning when the package contains both the PackageIcon and PackageIconUrl.
I've tested locally with my Cake.Recipe 2.0.0 build, no more warnings there.
And the [documentation](https://aka.ms/deprecateIconUrl) states that as long as the package has PackageIcon, you will not see the warning anymore, so I did not add the warning suppression as it's not needed.

Comment (augustoproiete) [reply src/Cake.Yaml/Cake.Yaml.csproj:17]: Excellent. Thanks @escarls !

**Review by augustoproiete — APPROVED** [review]
LGTM!

**Comment by augustoproiete** [comment]
@escarls your changes have been merged, thanks for your contribution :+1:

**Pull request merged**